stackslib/src/net/p2p.rs

```diff
@@ -5907,6 +5909,55 @@ impl PeerNetwork {
         Ok(())
     }
 
+    /// Static helper to check to see if there has been a reorg
+    pub fn is_reorg(
+        last_sort_tip: Option<&BlockSnapshot>,
+        sort_tip: &BlockSnapshot,
+        sortdb: &SortitionDB,
+    ) -> bool {
+        let reorg = if let Some(last_sort_tip) = last_sort_tip {
+            if last_sort_tip.block_height == sort_tip.block_height
+                && last_sort_tip.consensus_hash != sort_tip.consensus_hash
+            {
+                debug!(
+                    "Reorg detected at burn height {}: {} != {}",
+                    sort_tip.block_height, &last_sort_tip.consensus_hash, &sort_tip.consensus_hash
+                );
+                true
+            } else if last_sort_tip.block_height != sort_tip.block_height {
+                // last_sort_tip must be an ancestor
+                let ih = sortdb.index_handle(&sort_tip.sortition_id);
+                if let Ok(Some(ancestor_sn)) =
+                    ih.get_block_snapshot_by_height(last_sort_tip.block_height)
+                {
+                    if ancestor_sn.consensus_hash != last_sort_tip.consensus_hash {
+                        info!(
+                            "Reorg detected at burn block {}: ancestor tip at {}: {} != {}",
+                            sort_tip.block_height,
+                            last_sort_tip.block_height,
+                            &ancestor_sn.consensus_hash,
+                            &last_sort_tip.consensus_hash
+                        );
+                        true
+                    } else {
+                        false
+                    }
+                } else {
+                    info!(
+                        "Reorg detected: no ancestor of burn block {} ({}) found",
+                        sort_tip.block_height, &sort_tip.consensus_hash
+                    );
+                    true
+                }
+            } else {
+                false
+            }
+        } else {
+            false
+        };
```

Comment:
These checks should be inverted, immediate returned, and flattened... it would make following the logic here way easier.

Because it's very difficult to figure out what these cases correspond to:

```rust
                        true
                    } else {
                        false
                    }
                } else {
                    info!(
                        "Reorg detected: no ancestor of burn block {} ({}) found",
                        sort_tip.block_height, &sort_tip.consensus_hash
                    );
                    true
                }
            } else {
                false
            }
        } else {
            false
        };
```

Reply:
Sure, happy to.

Btw, I'm not doing this on purpose.  When I'm writing code like this, I'm envisioning a decision tree moreso than a checklist.  Each code body is one path through that tree.  But I appreciate that it creates a lot of avoidable indentation.